### PR TITLE
Retrieve SAS from a kv

### PR DESCRIPTION
**Pull request by nakulkar-msft**
* Set keyvault name and secret name in the plugin config file.
az_kv_name=""
az_kv_secret_name=""

* If using user-managed identity, set 'AZURE_CLIENT_ID' to the client id of user-managed identity.

* If retrieval of SAS from KV fails, we use the SAS fetched during previous update

* Key is refreshed by the first action (Archive/Restore/Remove) which occurs at least 24hrs (Not configurable) after previous refresh.

### CODE REVIEW

File: cmd/lhsm-plugin-az/main.go [hunk 1]
```diff
@@ -24,35 +24,39 @@ import (
...
-		Bandwidth        int    `hcl:"bandwidth"`
-		MountRoot        string `hcl:"mountroot"`
-		ExportPrefix     string `hcl:"exportprefix"`
-		azCreds          *azblob.SharedKeyCredential
+		Name                  string `hcl:",key"`
+		ID                    int
+		AzStorageAccount      string `hcl:"az_storage_account"`
+		AzStorageKVName       string `hcl:"az_kv_name"`
+		AzStorageKVSecretName string `hcl:"az_kv_secret_name"`
+		AzStorageSAS          string `hcl:"az_storage_sas"`
```
Comment (zezha-msft) [inline cmd/lhsm-plugin-az/main.go:32]: From our discussion `az_storage_sas` is no longer needed

File: cmd/lhsm-plugin-az/main.go [hunk 2]
```diff
@@ -94,8 +98,8 @@ func (a *archiveConfig) checkValid() error {
 }
 
 func (a *archiveConfig) checkAzAccess() error {
-	if _, err := azblob.NewSharedKeyCredential(a.AzStorageAccount, a.AzStorageKey); err != nil {
-		return errors.Wrap(err, "No Az credentials found; cannot initialize data mover")
+	if a.AzStorageKVName != "" && a.AzStorageKVSecretName != "" {
```
Comment (zezha-msft) [inline cmd/lhsm-plugin-az/main.go:101]: Shouldn't we return an error if there's no KV credentials?

**Review by zezha-msft — APPROVED** [review]

**Comment by zezha-msft** [comment]
@nakulkar-msft just to confirm my assumption, the SAS in the kv is updated every day, but it's valid for 2 days, correct? So even if we get a SAS out of kv right before it got rotated, it'll still be good enough for the next 24 hours.

File: cmd/lhsm-plugin-az/main.go (hunk 1 above)
Comment (nakulkar-msft) [reply cmd/lhsm-plugin-az/main.go:32]: I'll need that to hold the SAS key after we get it from kv. I'll remove the serialization to get it from config file.

**Comment by nakulkar-msft** [comment]
> @nakulkar-msft just to confirm my assumption, the SAS in the kv is updated every day, but it's valid for 2 days, correct? So even if we get a SAS out of kv right before it got rotated, it'll still be good enough for the next 24 hours.

Yes. At any point in time, the sas in kv is valid at least for 24hrs. It might happen that we dont reach kv for more than 2 days, but any time we reach, we get a SAS which is valid atleast for next 24 hrs.

**Pull request merged**